src/internal/internal_he2hb_gemm.cc

```diff
@@ -154,124 +156,106 @@ void he2hb_gemm(
-                int64_t mt = C.mt();
-
-                // check if there are cleanup tiles
-                if (C.tileMb( mt-1 ) != C.tileMb( 0 )) {
-                    i_interior = A.mt() - 1;
-                    i_last = 1;
-                }
-
-                // interior
-                std::vector<scalar_t*> a_array00;
-                std::vector<scalar_t*> b_array00;
-                std::vector<scalar_t*> c_array00;
-                a_array00.reserve( batch_size );
-                b_array00.reserve( batch_size );
-                c_array00.reserve( batch_size );
-
-                int64_t lda00 = 0;
-                int64_t ldb00 = 0;
-                int64_t ldc00 = 0;
-                int64_t mb00 = C.tileMb( 0 );
-                int64_t nb00 = C.tileNb( 0 );
-                int64_t kb   = A.tileNb( 0 );
-                for (int64_t i = 0; i < i_interior; ++i) {
-                    if (A.tileRank( i, k ) == panel_rank
-                        && device == C.tileDevice( i, 0 )) {
-                        a_array00.push_back( A( i, k, device ).data() );
-                        b_array00.push_back( B( k, 0, device ).data() );
-                        c_array00.push_back( C( i, 0, device ).data() );
-                        lda00 = A( i, k, device ).stride();
-                        ldb00 = B( k, 0, device ).stride();
-                        ldc00 = C( i, 0, device ).stride();
-                    }
-                }
-
-                // if mod( n, nb ) != 0, this is for the last tile
-                std::vector<scalar_t*> a_array11;
-                std::vector<scalar_t*> b_array11;
-                std::vector<scalar_t*> c_array11;
-                a_array11.reserve( batch_size );
-                b_array11.reserve( batch_size );
-                c_array11.reserve( batch_size );
-
-                int64_t lda11 = 0;
-                int64_t ldb11 = 0;
-                int64_t ldc11 = 0;
-                int64_t mb11 = C.tileMb( C.mt()-1 );
-                int64_t nb11 = C.tileNb( C.nt()-1 );
-                // same kb as above
-                {
-                    int i = C.mt()-1;
-                    if ((A.tileRank( i, k ) == panel_rank) && (i_last == 1)) {
-                        if (device == C.tileDevice( i, 0 )) {
-                            a_array11.push_back( A( i, k, device ).data() );
-                            b_array11.push_back( B( k, 0, device ).data() );
-                            c_array11.push_back( C( i, 0, device ).data() );
-                            lda11 = A( i, k, device ).stride();
-                            ldb11 = B( k, 0, device ).stride();
-                            ldc11 = C( i, 0, device ).stride();
+                scalar_t** a_array_host = host_work;
+                scalar_t** b_array_host = a_array_host + batch_size;
+                scalar_t** c_array_host = b_array_host + batch_size;
+
+                // Varient of device_regions_build to handle trsmA
```

Comment:
"trsmA"? This is gemm. Although it's not an outer product like gemmC, from my recollection.